pkg/controller/scyllacluster/resource.go

```diff
@@ -219,9 +221,25 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 	if err != nil {
 		return nil, fmt.Errorf("can't get rack labels: %w", err)
 	}
-	matchLabels := helpers.ShallowCopyMap(rackLabels)
+	_, rackOrdinal, ok := slices.Find(c.Spec.Datacenter.Racks, func(rack scyllav1.RackSpec) bool {
+		return rack.Name == r.Name
+	})
+	if !ok {
+		return nil, fmt.Errorf("can't find ordinal of rack %q in ScyllaCluster %q", r.Name, naming.ObjRef(c))
+	}
+	rackLabels[naming.RackOrdinalLabel] = strconv.Itoa(rackOrdinal)
 	rackLabels[naming.ScyllaVersionLabel] = c.Spec.Version
 
+	selectorLabels, err := naming.RackLabels(r, c)
```

Comment:
I think `RackLabels` might need to be renamed to `RackSelectorLabels` or something like that, so there is a clear distinction + a comment not to change these.

Reply:
done